lib/backend/k8s/resources/resources.go

```diff
@@ -100,3 +103,81 @@ func SetCalicoMetadataFromK8sAnnotations(calicoRes Resource, k8sRes Resource) {
 		}
 	}
 }
+
+// Store Calico Metadata in the in the k8s resource annotations for CRD backed resources.
+// This should store all Metadata except for those stored in Annotations and Labels and
+// store them in annotations.
+func StoreCalicoMetadataInK8sAnnotations(resIn Resource) (Resource, error) {
```

Comment:
Not sure about these function names.  Maybe these should just be ConvertCalicoResourceToK8sResource and ConvertK8sResourceToCalicoResource - obviously we still need to document what that conversion entails.